src/main/java/school/lemon/changerequest/java/generics/NumberGenerator/NumberGeneratorImp.java

```diff
@@ -0,0 +1,23 @@
+package school.lemon.changerequest.java.generics.NumberGenerator;
+
+/**
+ * Created by lera on 01.01.17.
+ */
+public class NumberGeneratorImp<T extends Number> extends NumberGenerator<T> {
+    T[] arrayT;
```

Comment:
For sure, number generator shouldn't store generated numbers. Please remove this field.